File: Easy/A1.py
# Name:- Prashanth V Gowda

# Date:- 17/11/2024

# Description:- Given a string find the first K non-repeating characters
# in it by doing only single traversal of it

# Example
# Input:- ABCDBAGHCHFAC or ABCDZBAGHCHFACBICJBKI and K=3 

# OUtput:- D G F

# Time:-
# Space:- 

stringValue = "ABCDZBAGHCHFACBICJBKI"
kvalue = 3
foundRepeats = 0

for x in range(len(stringValue)):
    if foundRepeats > 5:
        break
    foundChar = 0
    for y in range(len(stringValue)):
        if x != y and stringValue[x] == stringValue[y]:
            foundChar = 1
            break
    if foundChar == 0:
        print(stringValue[x])
        foundRepeats += 1
        

File: Easy/A1.c
/*
Name:- Prashanth V Gowda

Date:- 17/11/2024

Description:- Given a string find the first K non-repeating characters
in it by doing only single traversal of it

Example
Input:- ABCDBAGHCHFAC or ABCDZBAGHCHFACBICJBKI and K=3 

OUtput:- D G F

Time:-
Space:- 

*/

#include <stdio.h>
#include <string.h>

int main()
{
    char stringValue[] = "ABCDZBAGHCHFACBICJBKI";
    int stringLength = strlen(stringValue);
    int foundRepeats = 0;

    for (int i = 0; i < stringLength && foundRepeats < 3; i++)
    {
        int foundChar = 0;
        for (int j = 0; j < stringLength; j++)
        {
            if (i !=j && stringValue[i] == stringValue[j])
            {
                foundChar = 1;
                break;
            }
        }
        if (foundChar == 0)
        {
            printf("%c", stringValue[i]);
            foundRepeats++;
        }
    }
}

/*
    printf("%d \n", stringLength);
    int stringLength2  = sizeof(string) / sizeof(string[0]);
    printf("%d \n", stringLength2);
    printf("%d \n", sizeof(string[0]));
*/